### PR TITLE
TINY-6918: Switch silver theme to use BDD style tests

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6918

Description of Changes:
* Fixed an issue where the return type for the UI `onSetup` API wasn't typed correctly and improved it to check the return is a function, as previously it only checked it was nonnullable.
* Added new FocusTools helper to get the active value
* Switched all silver theme test to use BDD style tests

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] ~Branch prefixed with `feature/` for new features (if applicable)~
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by JamesToohey — COMMENTED** [review]
Leaving these comments from last week while I finish reviewing

File: modules/tinymce/src/themes/silver/test/ts/atomic/components/sizeinput/SizeInputFormatTest.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+describe('atomic.tinymce.themes.silver.components.sizeinput.SizeInputFormatTest', () => {
+  it('pixel measurements do not get decimals', () => {
+    assert.equal(formatSize({ value: 12, unit: 'px' }), '12px');
+    assert.equal(formatSize({ value: 12.123, unit: 'px' }), '12px');
+    assert.equal(formatSize({ value: 12.123, unit: '' }), '12');
+    assert.equal(formatSize({ value: 100.1, unit: '' }), '100');
+  });
+
+  it('percentages get up to 4 decimal places', () => {
+    assert.equal(formatSize({ value: 1.1234321, unit: '%' }), '1.1234%');
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/themes/silver/test/ts/atomic/components/sizeinput/SizeInputFormatTest.ts:15]: Possibly, but that's not related to the conversion of the tests.

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/ToolbarBottomTest.ts [hunk 2]
```diff
@@ -1,56 +1,53 @@
...
-  Theme();
+interface Scenario {
+  readonly settings: RawEditorSettings;
+  readonly initial: Array<{ clickOn: string; waitFor: string }>;
+  readonly assertAbove: string;
+  readonly assertBelow: string;
+}
+
+describe('browser.tinymce.themes.silver.editor.ToolbarBottomTest', () => {
+  before(() => {
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/ToolbarBottomTest.ts:22]: I did a quick search in this PR and I can't see any. I always prefer this style anyways, as the newline doesn't achieve anything.

**Review by JamesToohey — APPROVED** [review]

File: modules/tinymce/src/themes/silver/test/ts/module/StickyHeaderStep.ts [hunk 3]
```diff
@@ -1,63 +1,99 @@
-import { Assertions, GeneralSteps, Pipeline, Step, UiFinder, Waiter } from '@ephox/agar';
-import { Cell } from '@ephox/katamari';
-import { TinyApis, TinyLoader } from '@ephox/mcagar';
+import { UiFinder, Waiter } from '@ephox/agar';
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/themes/silver/test/ts/module/StickyHeaderStep.ts:1]: I considered that, but couldn't think of a good name. It's also still relevant, as it's testing the various steps/stages in the sticky header behaviour.

**Review by ltrouton — APPROVED** [review]

**Pull request merged**